tools/ExtensionLibrary/DevHome.ExtensionLibrary/ViewModels/ExtensionLibraryViewModel.cs

```diff
@@ -2,22 +2,151 @@
 // Licensed under the MIT license.
 
 using System;
+using System.Collections.ObjectModel;
 using System.Threading.Tasks;
 using CommunityToolkit.Mvvm.ComponentModel;
 using CommunityToolkit.Mvvm.Input;
+using CommunityToolkit.WinUI;
+using DevHome.Common.Extensions;
+using DevHome.Common.Services;
+using DevHome.Dashboard.Helpers;
+using Microsoft.UI.Xaml;
+using Windows.Data.Json;
+using Windows.Storage;
 using Windows.System;
 
 namespace DevHome.ExtensionLibrary.ViewModels;
 
 public partial class ExtensionLibraryViewModel : ObservableObject
 {
+    private readonly string devHomeProductId = "9N8MHTPHNGVV";
+
+    private readonly Microsoft.UI.Dispatching.DispatcherQueue _dispatcher;
+
+    [ObservableProperty]
+    private ObservableCollection<StorePackageViewModel> _storePackagesList = new ();
+
+    [ObservableProperty]
+    private bool _shouldShowStoreError = false;
+
     public ExtensionLibraryViewModel()
     {
+        _dispatcher = Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread();
+
+        var pluginService = Application.Current.GetService<IPluginService>();
+        pluginService.OnPluginsChanged -= OnPluginsChanged;
+        pluginService.OnPluginsChanged += OnPluginsChanged;
+
+        GetAvailablePackages();
     }
 
     [RelayCommand]
     public async Task GetUpdatesButtonAsync()
     {
         await Launcher.LaunchUriAsync(new ("ms-windows-store://downloadsandupdates"));
     }
+
+    private async void OnPluginsChanged(object? sender, EventArgs e)

```

Comment:
I thought we were moving away from the "Plugin" term towards more "Extension" term. Should this be OnExtensionsChanged?

Reply:
I figured we'd be doing a bigger pass to rename the events/IPluginService to say Extension and thought I would keep it consistent until that when we can rename the handlers too. I don't feel super strongly about it.